File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0;
    function dfs(i, result) {
        if (i===numbers.length) {
            if (result===target) answer++;
            return;
        }
        else {
            dfs(i+1, result+numbers[i]);
            dfs(i+1, result-numbers[i]);
        }
    }
    dfs(0, 0);
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    global cnt
    cnt = 0
    
    def dfs(idx, tot):
        global cnt
        if idx==len(numbers)-1:
            if tot==target:
                cnt+=1
            
        else:
            dfs(idx+1, tot+numbers[idx+1])
            dfs(idx+1, tot-numbers[idx+1])
            
    dfs(-1,0)
    return cnt